src/promptflow/promptflow/_sdk/_service/apis/ui.py

```diff
@@ -16,3 +38,46 @@ def get(self):
             render_template("index.html", url_for=url_for),
             mimetype="text/html",
         )
+
+
+def save_image(base64_data, extension):
+    image_data = base64.b64decode(base64_data)
+    filename = str(uuid.uuid4())
+    file_path = HOME_PROMPT_FLOW_DIR / f"{filename}.{extension}"
+    with open(file_path, "wb") as f:
+        f.write(image_data)
+    return str(file_path)
+
+
+@api.route('/media_save')
+class MediaSave(Resource):
+    @api.response(code=200, description="Save image", model=fields.String)
+    @api.doc(description="Save image")
+    @api.expect(media_save_model)
+    def post(self):
+        base64_data = api.payload["base64_data"]
+        extension = api.payload["extension"]
+        file_path = save_image(base64_data, extension)
+        return file_path
+
+
+@api.route('/image')
+class ImageUrl(Resource):
+    @api.response(code=200, description="Get image url", model=fields.String)
+    @api.doc(description="Get image url")
+    @api.expect(image_view_model)
+    def get(self):
+        image_path = api.payload["image_path"]
+        if not os.path.exists(image_path):
+            return make_response("The image doesn't exist", 404)
+        directory, filename = os.path.split(image_path)
+        directory = Path(directory).as_posix()
+        return f"{request.base_url}/{str(directory)}/{filename}"
+
+
+@api.route('/image/<path:directory>/<path:filename>')
+class ImageView(Resource):
+    @api.doc(description="Visualize image")
+    @api.response(code=200, description="Visualize image", model=fields.String)
+    def get(self, directory, filename):
+        return send_from_directory(directory, filename)
```

Comment:
Can this url be view in brower if this is a image?

Reply:
yeah,
![image](https://github.com/microsoft/promptflow/assets/26239730/3c56c5bf-11d3-4661-94f6-12e4da505e5e)